### PR TITLE
feature: [PAR-4351] add post purchase lead modal

### PR DESCRIPTION
**Pull request by milesmcleod**
# Description

* add new template to pass parameters to `aftermarketProductProtectionModalOffer`
* add new js file to handle opening the aftermarket modal
* add new function `getLeadTokenFromUrl()` to parse the lead token from the query parameters
* update ProductProtection.php - i was running into an issue where the item added to the quote was not being correctly persisted, and from my digging it seemed like it was because the quoteId was being set on it. i believe the way it is now works because
  * the item is added to the quote and persisted
  * then when the quote is reloaded, the item is present on it and `collectTotals` will run
    * if the quote is not fetched again after it is persisted with the item, then `collectTotals` will not run - there is a boolean flag that gets set when it is reloaded, which i assume is because a new item has been loaded and the existing total does not match.

## Testing
* add a valid lead token to the cart by itself and check out -> contract is created, lead token is `consumed` ✅
* add a valid lead token to the cart, then add other items and check out -> contract is created, lead token is `consumed` ✅
* add other items, then add a lead token to the cary -> contract is created, lead token is `consumed` ✅

## Questions
* ~i noticed error `Missing required option "storeId"` occasionally occurs (in the browser console) when navigating to the store with a lead token in the query string, after clearing cookies/the cache. this is an error caused by the aftermarket modal trying to open before `'storeId'` is set in the sdk's `globalConfig` - not sure what the right resolution is there, but it's a question i had.~
  * fixed by explicitly passing the storeId into the new js function, setting to Extend.config, and explicitly passing it to the sdk call to open the modal

## Ticket

https://helloextend.atlassian.net/browse/PAR-4351

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by scott-taylor-extend — APPROVED** [review]

File: view/frontend/layout/default.xml [hunk 1]
```diff
@@ -11,6 +11,11 @@
                     <argument name="viewModel" xsi:type="object">Extend\Integration\ViewModel\EnvironmentAndExtendStoreUuid</argument>
                 </arguments>
             </block>
+            <block name="extend.aftermarket-product-protection-modal-offer" ifconfig="extend/product_protection/enable" template="Extend_Integration::catalog/product/view/aftermarket-product-protection-modal-offer.phtml">
+                <arguments>
+                    <argument name="viewModel" xsi:type="object">Extend\Integration\ViewModel\EnvironmentAndExtendStoreUuid</argument>
+                </arguments>
+            </block>
```
Comment (jay-faust-extend) [inline view/frontend/layout/default.xml:18]: Why is this added here versus its own file? (Not arguing it should be its own file, I'm just curious)

**Review by jay-faust-extend — APPROVED** [review]

File: view/frontend/layout/default.xml (hunk 1 above)
Comment (milesmcleod) [reply view/frontend/layout/default.xml:18]: i placed it here because i believe this layout applies on every page, so regardless of what the base URL is, if the leadToken is present, the modal will work.

guidance puts it in `cms_index_index.xml` which would probably work for us too, but we haven't created that file yet, and i don't think this is too different. i didn't think too hard about this but we could definitely revisit if we decide it would be better elsewhere!
